plugin/src/main/groovy/com/novoda/staticanalysis/internal/lint/LintConfigurator.groovy

```diff
@@ -0,0 +1,63 @@
+package com.novoda.staticanalysis.internal.lint
+
+import com.novoda.staticanalysis.StaticAnalysisExtension
+import com.novoda.staticanalysis.internal.Configurator
+import com.novoda.staticanalysis.internal.Violations
+import org.gradle.api.NamedDomainObjectContainer
+import org.gradle.api.Project
+import org.gradle.api.Task
+
+class LintConfigurator implements Configurator {
+
+    private final Project project
+    private final Violations violations
+    private final Task evaluateViolations
+
+    static LintConfigurator create(Project project,
+                                   NamedDomainObjectContainer<Violations> violationsContainer,
+                                   Task evaluateViolations) {
+        Violations violations = violationsContainer.maybeCreate('Lint')
+        return new LintConfigurator(project, violations, evaluateViolations)
+    }
+
+    private LintConfigurator(Project project, Violations violations, Task evaluateViolations) {
+        this.project = project
+        this.violations = violations
+        this.evaluateViolations = evaluateViolations
+    }
+
+    @Override
+    void execute() {
+        project.extensions.findByType(StaticAnalysisExtension).ext."lintOptions" = { Closure config ->
+            if (!isAndroidProject(project)) {
+                return
+            }
+
+            project.android.lintOptions(config)
```

Comment:
This seems to be missing the bit where we avoid lint to break the build (we want to be in charge of when doing that). I believe we should force `abortOnError false` here.

Reply:
Having a look at the whole options here: https://google.github.io/android-gradle-dsl/current/com.android.build.gradle.internal.dsl.LintOptions.html

I think we can also add `xmlReport true` to make sure that xml report is enabled.

Reply:
Same for the htmlReport I'd say